userspace/engine/json_evt.h

```diff
@@ -59,41 +60,100 @@ class json_event : public gen_event
 	uint64_t m_event_ts;
 };
 
+// A class representing an extracted value or a value on the rhs of a
+// filter_check. This intentionally doesn't use the same types as
+// ppm_events_public.h to take advantage of actual classes instead of
+// lower-level pointers pointing to syscall events and to allow for
+// efficient set comparisons.
```

Comment:
Cool!

Could you just document the logic of comparisons. I'm referring to the `==`, `!=`, `<`, `>` operators I see below. Looking at the implementation the logic is clear to me. Nevertheless would be preferable to document it here in plain words for everyone :)

Reply:
Sure, done.